src/_pytest/scope.py

```diff
@@ -0,0 +1,86 @@
+"""
+Scope definition and related utilities.
+
+Those are defined here, instead of in the 'fixtures' module because
+their use is spread across many other pytest modules, and centralizing it in 'fixtures'
+would cause circular references.
+
+Also this makes the module light to import, as it should.
+"""
+from enum import Enum
+from typing import Optional
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing_extensions import Literal
+
+    _ScopeName = Literal["session", "package", "module", "class", "function"]
+
+
+class Scope(Enum):
+    """
+    Represents one of the possible fixture scopes in pytest.
+
+    Scopes are ordered from lower to higher, that is:
+
+              ->>> higher ->>>
+
+    Function < Class < Module < Package < Session
+
+              <<<- lower  <<<-
+    """
+
+    # Scopes need to be listed from lower to higher.
+    Function: "_ScopeName" = "function"
+    Class: "_ScopeName" = "class"
+    Module: "_ScopeName" = "module"
+    Package: "_ScopeName" = "package"
+    Session: "_ScopeName" = "session"
+
+    def next_lower(self) -> "Scope":
+        """Return the next lower scope."""
+        scopes = list(Scope)
+        index = scopes.index(self)
+        if index == 0:
+            raise ValueError(f"{self} is the lower-most scope")
+        return scopes[index - 1]
+
+    def next_higher(self) -> "Scope":
+        """Return the next higher scope."""
+        scopes = list(Scope)
+        index = scopes.index(self)
+        if index == len(scopes) - 1:
+            raise ValueError(f"{self} is the upper-most scope")
+        return scopes[index + 1]
+
+    def __lt__(self, other: "Scope") -> bool:
```

Comment:
We should apply `@functools.total_ordering` to the class so that all of the comparisons are implemented.

Reply:
Oops completely missed that, thanks!